text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+
+## Goals
+
+This proposal will enhance cargo and crates.io through adoption of the Sigstore capabilities and workflows for supporting both use cases as outlined above. Specifically, 
+
+* Adopt Sigstore infrastructure to allow for the signing of crates for use both above.
+* Facilitate workflows in both Cargo and crates.io to allow for signature generation in the former case and verification in the latter case of crates and their dependencies by crate consumers (a note on dependencies: the depth level should be 1 for verification in this first phase)
+
+This implementation will support all three Sigstore implementation models which will be effectively transparent to both crates.io maintainers and crates.io consumers. At a high level the three Sigstore implementation models are:
+
+1. Use of the public Sigstore common good service, signing with the Sigstore public root.
+1. Signature operations via a private fulcio instance with intermediate signer issued internally, but signature metadata pushed to public Rekor instance.
+1. Full internal Sigstore infra used with internal consumption only software.
+
+# Non Goals
+
+* Implementing TUF - this proposal describes a sequenced implementation with TUF in a phase 2. 
+* Signing crates.io index - this is covered by other proposals, but is discussed in the context of TUF later in this document.
+* Extending or altering the Identity approach of Sigstore, i.e. OIDC based
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Crate signatures and verifications are a way to tie the authenticity and integrity of a crate based on public key cryptography. The implication is that, as an end user, you can be sure that the crates that you download from crates.io have not been tampered with, and that they are signed by the correct owner of the package.
+
+The process of `signing` involves generating a signature for a crate, using a private key, and attaching that signature when publishing a crate. The process of `verification` involves downloading a crate (using cargo) and using a _trusted_ public key to verify the authenticity (the signer) and integrity (the contents) of a crate.
+
+To simplify key management for users, cargo supports using [Sigstore](https://sigstore.dev) as a way to generate ephemeral keys based on an identity, and as an immutable log that can be monitored and audited in the event of a compromise. By default, a publicly available instance ensures that you do not need to manage this infrastructure yourself. If you are running your own crate registry, Sigstore instance and identify provider, you can configure cargo to use that.
+
+Verification is accomplished through an API interaction with the relevant Rekor service. This occurs at artifact submission time to the relevant crates registry and is also how the consumer will verify the artifact whether manually or in a run-time scenario.
+
+## Implementation sequencing
```

Comment:
This entire section feels not guide-level. Is the end functionality for users (e.g. available cargo commands) changed between any of the phases? That's what's worthwhile to mention here, otherwise, it goes in the next section.